### PR TITLE
add ITS online FEE task

### PR DESCRIPTION
**Pull request by JianLIUhep**
Looking into the ITS diagnostic payload to spot the detector errors.

### CODE REVIEW

**Comment by Barthelemy** [comment]
the failure is spurious

File: Modules/ITS/src/ITSFeeTask.cxx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    if (mEnableLayers[ilayer]) {
+      ILOG(Info) << "Enable layer : " << ilayer << ENDM;
+    }
+  }
+  configFile >> mRunNumberPath;
+}
+
+void ITSFeeTask::endOfCycle()
+{
+  std::ifstream runNumberFile("/home/its/QC_Online/workdir/infiles/RunNumber.dat"); //catching ITS run number in commissioning
```
Comment (Barthelemy) [inline Modules/ITS/src/ITSFeeTask.cxx:167]: that is likely to break on any other machine than yours

File: Modules/ITS/src/ITSFeeTask.cxx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  if (runNumberFile) {
+    std::string runNumber;
+    runNumberFile >> runNumber;
+    ILOG(Info) << "runNumber : " << runNumber << ENDM;
+    if (runNumber == mRunNumber) {
```
Comment (Barthelemy) [inline Modules/ITS/src/ITSFeeTask.cxx:172]: perhaps `if (runNumber != mRunNumber){
    getObjectsManager()->addMetadata(mTFInfo->GetName(), "Run", runNumber);
    getObjectsManager()->addMetadata(mErrorFlagVsFeeId->GetName(), "Run", runNumber);
    getObjectsManager()->addMetadata(mErrorFlag->GetName(), "Run", runNumber);
    mRunNumber = runNumber;
}
I think that it is logically the same but it avoids the `goto`.

**Review by Barthelemy — COMMENTED** [review]
Very good, I have just a couple of comments. Could you consider them ? Cheers

File: Modules/ITS/src/ITSFeeTask.cxx (hunk 1 above)
Comment (JianLIUhep) [reply Modules/ITS/src/ITSFeeTask.cxx]: Amended. Passing the run number through the cfg for ITS commissioning.

File: Modules/ITS/src/ITSFeeTask.cxx (hunk 2 above)
Comment (JianLIUhep) [reply Modules/ITS/src/ITSFeeTask.cxx]: removed. no need to make the checking here.

**Review by Barthelemy — APPROVED** [review]

**Pull request merged**